sendgrid.go

```diff
@@ -26,19 +26,38 @@ type SGClient struct {
 }
 
 // NewSendGridClient will return a new SGClient.
-func NewSendGridClient(apiUser, apiPwd string) *SGClient {
+func NewSendGridClient(apiUserOrKey string, apiKey ...string) *SGClient {
```

Comment:
@eddiezane - I would also recommend having two separate `constructors`:  one for user/password style clients and another for api user or key.  This seams to possibly overload the functionality of the single NewSendGridClient a little too much.  (Just a suggestion.)